src/components/raycaster.js

```diff
@@ -31,6 +31,15 @@ module.exports.Component = registerComponent('raycaster', {
     this.prevCheckTime = undefined;
     this.raycaster = new THREE.Raycaster();
     this.updateOriginDirection();
+    this.refreshObjects = this.refreshObjects.bind(this);
+  },
+
+  play: function () {
+    this.el.sceneEl.addEventListener('child-attached', this.refreshObjects);
```

Comment:
We would have to run `el.match` and `el.querySelectorAll` on the child and remove one by one all the query matches when a child is detached. For simplicity sake I prefer to just `refreshObjects` on attach / detach and make it more granular later if needed.